src/main/java/org/terasology/module/health/events/BeforeRegenEvent.java

```diff
@@ -4,17 +4,31 @@
 package org.terasology.module.health.events;
 
 import org.terasology.engine.entitySystem.event.AbstractConsumableValueModifiableEvent;
+import org.terasology.naming.Name;
 
+/**
+ * A <i>collector event</i> to allow interested systems to contribute to a regeneration action.
+ *
+ * A separate collector event is sent for each registered regeneration id. The resulting amount per regeneration id is
+ * at minimum 0 (no negative effects due to regeneration). The final regeneration amount is computed from the sum of
+ * all amounts per regeneration id.
+ *
+ * To inflict damage caused by regeneration send a separate {@link DoDamageEvent} instead.
+ */
 public class BeforeRegenEvent extends AbstractConsumableValueModifiableEvent {
 
-    private final String id;
+    final Name id;
```

Comment:
why is this not `private`?

Reply:
Good point. I'll incorporate that in one of the follow-up PRs if that's okay with you.

Reply:
yes, okay